### DISCUSSION TITLE
Let's start a discussions here

### DISCUSSION THREAD

**Discussion by mukeenanyafiq** [aieuo/Mineflow#137] [General]
### Question

Let's start a discussions here
Ask anything or discuss everything in here about plugins and others

**Comment by mukeenanyafiq** [comment]
so, if we add a new command, it's added. but it isn't listed in command list. does the plugin works like that?

**Comment by aieuo** [comment]
I think that will be listed after restarting the server or rejoining the server.

**Comment by RayanK0** [comment]
Any idea how to make it so you can only execute a recipe 1 time?

**Comment by aieuo** [comment]
> Any idea how to make it so you can only execute a recipe 1 time?

One way I think is to set the flag using a global variable and check it at the beginning of the recipe.

**Comment by RayanK0** [comment]
how? Idk how to do it

**Comment by aieuo** [comment]
I made a [recipe](https://github.com/aieuo/MineflowExamples/blob/master/Counter.json) to implement it simply.  It works with the [latest build](https://poggit.pmmp.io/r/157764/Mineflow_dev-239.phar).
You can use the recipes, `Callback the other recipe.` action, `If` action, `Exit the recipe` action, and a `Compare the numbers.` condition like this:
![image](https://user-images.githubusercontent.com/35859665/150668428-5aef5945-07e6-4fc5-bcd5-88a566fa44db.png)
![image](https://user-images.githubusercontent.com/35859665/150668456-232f9a42-95c6-4c54-8a41-b4783aa9ed24.png)

The `<args>` of the second image is to identify the count. You can edit it, but you need to specify two values separated by a comma.

**Comment by RayanK0** [comment]
Call back the other recipe action is gone? I cant find it anymore, and where is the particles action?

**Comment by aieuo** [comment]
Run `/mineflow permission <your name> 2` from the console and you can see the actions.

**Comment by RayanK0** [comment]
I already have my permission to 2
I tried it again and still wont work, reinstalling the plugin wont work either also tried it on a other server and I still dont see the actions.

**Comment by aieuo** [comment]
I'm sorry for the late reply.  I tried it on my server and the action is listed below the `Wait`.  
Please open the `plugin_data/Mineflow/player.yml` and make sure your full name is listed like this:
![image](https://user-images.githubusercontent.com/35859665/151645028-b345531d-5810-4e7a-8960-050bada7c293.png)
![image](https://user-images.githubusercontent.com/35859665/151644850-01b047d6-1153-4c46-a8a7-975956c7ceac.png)

**Comment by RayanK0** [comment]
No worries
My name wasnt listed in te file, I did it manually and it works now!
I think something in my server is causing it to not work.
Thanks for helping

**Comment by mukeenanyafiq** [comment]
@aieuo Do you have any discord server for this plugin? If doesn't, try make a new discord server for this plugin or yours and add it into README for a link discord server

**Reply by aieuo** [reply]
I don't think it is necessary because we can use the issues and the [discussions](https://github.com/aieuo/Mineflow/discussions) on Github

**Comment by mukeenanyafiq** [comment]
I may can became a Translator role in your discord server. If you want to

**Comment by mukeenanyafiq** [comment]
and last, maybe I can now set-up a PMMP Server 4.0.0 like normally now

**Comment by RayanK0** [comment]
I would love a discord server for Mineflow.

**Comment by mukeenanyafiq** [comment]
oh my gosh, i thought that this issue was closed. it turns out to be the discussions

**Comment by mukeenanyafiq** [comment]
just realized, @aieuo using Windows 11

**Reply by aieuo** [reply]
yeah :D

**Reply by mukeenanyafiq** [reply]
how did you install that? are your system requirements is meet the minimum?

**Reply by aieuo** [reply]
I had been using insider preview before windows 11 was released and was automatically updated to windows 11 after it was released. 
I can't remember if there was a confirmation, but probably my system meets the minimum requirements.

**Reply by mukeenanyafiq** [reply]
that is my dream os for the future because it can install App from APKs file without Emulator 
maybe use Virtual Machine?

**Reply by mukeenanyafiq** [reply]
(yeah maybe better use virtual machine)

**Comment by TienDat2k** [comment]
no trigger when player level up? If not, is there a way to do it with a conditional statement? I tried but xp_level value doesn't change, how do I make a conditional statement that compares xp_level and gives "you leveled up" message in chat"

**Reply by aieuo** [reply]
There is no trigger for that, but I think you can create the following recipe to implement that.

### call trigger
Event trigger `When a player joins a server`
- `Add a variable` (xp, {target.xp_level}, number, local)
- `Repeat for every specified seconds while conditions are match`
  - condition `A player is online`
  - `if`
    - condition `Compare the numbers` ({target.xp} != {xp})
    - `Fire a custom trigger` (xp_level_up)
  - `Add a variable` (xp, {target.xp_level}, number, local)


### use trigger
Custom trigger `xp_level_up`
* `Send message to chat field` (you leveled up)

**Comment by mukeenanyafiq** [comment]
when the new version release?
or can i just try it the dev build?

**Comment by mukeenanyafiq** [comment]
@aieuo, what makes in your plugin have 1 error: "PHPStan Error"?

**Reply by aieuo** [reply]
I have no idea...

**Comment by mukeenanyafiq** [comment]
guys, there's a new discussions category called [Polls](https://github.com/aieuo/Mineflow/discussions/categories/polls) in GitHub
maybe with this, @aieuo can add a new features with Polls category

**Reply by mukeenanyafiq** [reply]
or has it existed last year??

**Comment by mukeenanyafiq** [comment]
it's been 4 months since this issue (now became duscussions) is created.
now i have an exam, it will start in 30 may
see you in 4 june

**Reply by mukeenanyafiq** [reply]
i'm back
maybe some of my exams need fixes

**Comment by mukeenanyafiq** [comment]
hi.

**Reply by aieuo** [reply]
hi :)

**Comment by mukeenanyafiq** [comment]
hi guys
@aieuo no updates?

**Comment by mukeenanyafiq** [comment]
@aieuo are you starting to use Crowdin as translation page?

**Reply by mukeenanyafiq** [reply]
i think i found it. [this is the project crowdin link](https://crowdin.com/project/mineflow)

**Reply by aieuo** [reply]
yes

I've granted you Indonesian proofreader permission at crowdin.

**Comment by natoriumm** [comment]
When I make group name ,I can't use '/'. Is there any solution?

**Reply by aieuo** [reply]
Please use the latest dev build from here https://poggit.pmmp.io/ci/aieuo/Mineflow/Mineflow

**Comment by mukeenanyafiq** [comment]
It's been 11 months since this discussion was created. And now, we are 2 months away from getting into 2023.

**Reply by mukeenanyafiq** [reply]
Well, previously it was a issue now it's a discussions

**Comment by mukeenanyafiq** [comment]
![Remote GitHub Repositories](https://user-images.githubusercontent.com/93072673/206471861-282df5a4-dc40-48ff-8d2c-644bed8a19f0.png)
i just installed an extensions from Visual Studio Code "GitHub Repositories" made by Official GitHub that could access to a repository remotely just in Visual Studio Code

**Reply by aieuo** [reply]
It looks useful

**Reply by mukeenanyafiq** [reply]
It kind of is but I never tried to pull commit yet with this extension. You can try it out by yourself

**Comment by mukeenanyafiq** [comment]
I'm back. It's been a while though. I'm about to graduate school next month. It has been turned into a great memories, that I will never forget. And this plugin too. First started when translating Indonesian language, now it turned into a whole crowdin page. I didn't approve some of the sentence yet. But I will. @aieuo, if you can, update your plugin. It's been 4 months. That's all.

**Reply by aieuo** [reply]
Congratulations on graduating from school next month and thank you very much for your translation into Indonesian Language.
I am currently working on creating API documentation for Mineflow3 and supporting for the pm4 economy plugin. Once that is completed, I plan to release Mineflow3.

**Comment by mukeenanyafiq** [comment]
Hey @aieuo, I have learned the basics of PHP thanks to the JavaScript and any sources that helps me. I can now make PocketMine-MP plugins for a server or for everyone that wanted to use the plugin. I successfully created a plugin manager plugin that only gives you an information about a plugin but only for my server and even updating someone's (published) plugin ([PlayerInfo](https://github.com/Matthww/PlayerInfo)) to support PMMP 5 even though my code is bad (right now he didn't commit my pull yet). I wanted to make a plugin for everyone that wanted to use but the problem is, I have no ideas for the purposes of the plugins. What kind of plugin do you think I should create?

**Reply by aieuo** [reply]
That's great! Unfortunately, I'm also out of ideas for new plugins :_(. You might be able to find some ideas on the plugin-ideas channel on the PMMP Discord server. https://discord.com/channels/373199722573201408/1020370001108152400

**Comment by mukeenanyafiq** [comment]
Just found out how to join a server on the same computer the server was hosted on in Windows.
Gotta try this plugin (Mineflow)

**Reply by mukeenanyafiq** [reply]
[this is how](https://doc.pmmp.io/en/rtfd/faq/connecting/win10localhostcantconnect.html)

**Comment by mukeenanyafiq** [comment]
It's been 6 months.
Hi there, @aieuo. How have you've been doing?
